enterprise/cmd/repo-updater/internal/authz/perms_syncer.go

```diff
@@ -683,6 +696,16 @@ func (s *PermsSyncer) syncUserPerms(ctx context.Context, userID int32, noPerms b
 		log.Object("fetchOpts", log.Bool("InvalidateCache", fetchOpts.InvalidateCaches)),
 	)
 
+	metricsSuccessPermsSyncs.WithLabelValues("user", string(p.UserID)).Inc()
+	metricsPermsFound.WithLabelValues("user", string(p.UserID)).Set(float64(len(p.IDs)))
+
+	if &oldPerms.SyncedAt != nil {
```

Comment:
`SyncedAt` is never going to be `nil`, so we will always end up not going in the other branch where we store the first sync metrics. See this code: https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@b104de5586c4ca6e1819ecb3d6e54351f06be81a/-/blob/enterprise/internal/database/perms_store.go?L1288&subtree=true

Same is true for the repositories.
```suggestion
	if !oldPerms.SyncedAt.IsZero() {
```

Reply:
updated